### PR TITLE
[INFRA/CORE] Add getNonDeadClientIDs and sort ClientID functions

### PR DESCRIPTION
**Pull request by lhl2617**
# Summary

As title. We need access to non dead clients often so make a function for that.

## Additional Info
changed @Eirikalb's stuff to use this

## Test Plan

Added unit tests boi

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

**Review by QFSW — APPROVED** [review]

File: internal/server/iifo.go [hunk 1]
```diff
@@ -73,13 +73,12 @@ func (s *SOMASServer) distributePredictions(islandPredictionDict shared.Predicti
...
-		if ci.LifeStatus != shared.Dead {
-			c := s.clientMap[id]
-			err = c.ReceivePredictions(receivedPredictionsDict[id])
-			if err != nil {
-				return errors.Errorf("Failed to receive prediction from client %v: %v", id, err)
-			}
+	nonDeadClients := getNonDeadClientIDs(s.gameState.ClientInfos)
+	for _, id := range nonDeadClients {
+		c := s.clientMap[id]
+		err = c.RecievePredictions(recievedPredictionsDict[id])
```
Comment (Darrekt) [inline internal/server/iifo.go]: typo in `recievedPredictionsDict`, and `c.RecievePredictions`. Checking if this exists anywhere else.

Comment (Darrekt) [reply internal/server/iifo.go]: Yall merged it while I was reviewing :(

Comment (lhl2617) [reply internal/server/iifo.go]: #94 fixed

**Pull request merged**